tsdb/engine/tsm1/writer.go

```diff
@@ -264,9 +264,10 @@ func NewDirectIndex() TSMIndex {
 // directIndex is a simple in-memory index implementation for a TSM file.  The full index
 // must fit in memory.
 type directIndex struct {
-	mu     sync.RWMutex
-	size   uint32
-	blocks map[string]*indexEntries
+	mu               sync.RWMutex
+	size             uint32
+	blocks           map[string]*indexEntries
+	cachedSortedKeys []string
```

Comment:
I think there needs to be a way to not have to resort the keys when calling Keys() every time. what do you think?